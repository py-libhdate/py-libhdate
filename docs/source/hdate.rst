hdate package
=============

Submodules
----------

hdate.common module
-------------------

.. automodule:: hdate.common
   :members:
   :undoc-members:
   :show-inheritance:

hdate.converters module
-----------------------

.. automodule:: hdate.converters
   :members:
   :undoc-members:
   :show-inheritance:

hdate.date module
-----------------

.. automodule:: hdate.date
   :members:
   :undoc-members:
   :show-inheritance:

hdate.htables module
--------------------

.. automodule:: hdate.htables
   :members:
   :undoc-members:
   :show-inheritance:

hdate.zmanim module
-------------------

.. automodule:: hdate.zmanim
   :members:
   :undoc-members:
   :show-inheritance:

Module contents
---------------

.. automodule:: hdate
   :members:
   :undoc-members:
   :show-inheritance:
