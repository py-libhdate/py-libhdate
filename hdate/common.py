"""Small helper classes."""

from dataclasses import dataclass
from datetime import tzinfo

import pytz

from hdate.htables import Months


class BaseClass:
    """Implement basic functionality for all classes."""

    def __eq__(self, other):
        """Override equality operator."""
        if isinstance(other, self.__class__):
            return self.__dict__ == other.__dict__
        return False

    def __ne__(self, other):
        """Override inequality operator."""
        return not self.__eq__(other)


@dataclass
class HebrewDate(BaseClass):
    """Define a Hebrew date object."""

    year: int
    month: Months
    day: int

    def __post_init__(self):
        self.month = (
            self.month if isinstance(self.month, Months) else Months(self.month)
        )


class Location(BaseClass):
    """Define a geolocation for Zmanim calculations."""

    # pylint: disable=too-many-arguments
    def __init__(
        self,
        name="Jerusalem",
        latitude=31.778,
        longitude=35.235,
        timezone="Asia/Jerusalem",
        altitude=754,
        diaspora=False,
    ):
        """Initialitze the location object."""
        self._timezone = None
        self.name = name
        self.latitude = latitude
        self.longitude = longitude
        self.timezone = timezone
        self.altitude = altitude
        self.diaspora = diaspora

    def __repr__(self):
        """Return a representation of Location for programmatic use."""
        return (
            f"Location(name='{self.name}', latitude={self.latitude}, "
            f"longitude={self.longitude}, timezone='{self.timezone}', "
            f"altitude={self.altitude}, diaspora={self.diaspora})"
        )

    @property
    def timezone(self):
        """Return the timezone."""
        return self._timezone

    @timezone.setter
    def timezone(self, value):
        """Set the timezone."""
        self._timezone = value if isinstance(value, tzinfo) else pytz.timezone(value)
