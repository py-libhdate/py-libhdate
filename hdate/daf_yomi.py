"""Daf Yomi module."""

import datetime as dt
from dataclasses import dataclass
from typing import ClassVar

from hdate.gematria import hebrew_number
from hdate.translator import TranslatorMixin


@dataclass(frozen=True)
class Masechta(TranslatorMixin):
    """Masechta object."""

    name: str
    pages: int

    def __str__(self) -> str:
        name = self.get_translation(self.name)
        daf = hebrew_number(self.pages + 2, language=self._language, short=True)
        return f"{name} {daf}"


@dataclass
class DafYomiDatabase:
    """Database of Masechtos."""

    # The first few cycles were only 2702 blatt. After that it became 2711. Even with
    # that, the math doesn't play nicely with the dates before the 11th cycle :(
    # From cycle 11 onwards, it was simple and sequential
    _start_date: ClassVar[dt.date] = dt.date(1997, 9, 29)
    _masechtot: ClassVar[list[Masechta]]

    @classmethod
    def register(cls, masechtot: list[Masechta]) -> None:
        """Regisetr a list of masechtos."""
        cls._masechtot = masechtot

    @classmethod
    def cycle_length(cls) -> int:
        """Return the length of a full cycle."""
        return sum(masechta.pages for masechta in cls._masechtot)

    def lookup(self, date: dt.date) -> Masechta:
        """Return the start date of a full cycle."""
        days_since_start = (date - self._start_date).days
        page_number = days_since_start % self.cycle_length()

        masechta_index = 0
        for masechta in self._masechtot:
            if page_number < masechta.pages:
                break
            page_number -= masechta.pages
            masechta_index += 1

        return Masechta(self._masechtot[masechta_index].name, page_number)


DAF_YOMI_MESECHTOS = (
    Masechta("berachos", 63),
    Masechta("shabbos", 156),
    Masechta("eruvin", 104),
    Masechta("pesachim", 120),
    Masechta("shekalim", 21),
    Masechta("yoma", 87),
    Masechta("succah", 55),
    Masechta("beitzah", 39),
    Masechta("rosh_hashanah", 34),
    Masechta("taanis", 30),
    Masechta("megillah", 31),
    Masechta("moed_katan", 28),
    Masechta("chagigah", 26),
    Masechta("yevamos", 121),
    Masechta("kesubos", 111),
    Masechta("nedarim", 90),
    Masechta("nazir", 65),
    Masechta("sotah", 48),
    Masechta("gittin", 89),
    Masechta("kiddushin", 81),
    Masechta("bava_kamma", 118),
    Masechta("bava_metzia", 118),
    Masechta("bava_basra", 175),
    Masechta("sanhedrin", 112),
    Masechta("makkos", 23),
    Masechta("shevuos", 48),
    Masechta("avodah_zarah", 75),
    Masechta("horayos", 13),
    Masechta("zevachim", 119),
    Masechta("menachos", 109),
    Masechta("chullin", 141),
    Masechta("bechoros", 60),
    Masechta("arachin", 33),
    Masechta("temurah", 33),
    Masechta("kereisos", 27),
    Masechta("meilah", 36),
    Masechta("niddah", 72),
)

DafYomiDatabase.register(list(DAF_YOMI_MESECHTOS))
