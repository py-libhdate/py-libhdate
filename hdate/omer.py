"""Omer module."""

from dataclasses import dataclass
from datetime import timedelta
from enum import Enum, auto
from typing import Union

from num2words import lang_HE, num2words

from hdate.gematria import hebrew_number
from hdate.hebrew_date import HebrewDate, Months
from hdate.translator import TranslatorMixin


class Nusach(Enum):
    """Nusach enum."""

    ASHKENAZ = auto()
    SFARAD = auto()
    ADOT_MIZRAH = auto()
    ITALIAN = auto()


@dataclass
class Omer(TranslatorMixin):
    """Hold information about the Omer count."""

    date: Union[HebrewDate, None] = None
    total_days: int = 0
    day: int = 0
    week: int = 0

    nusach: Nusach = Nusach.SFARAD
    language: str = "hebrew"

    def __post_init__(self) -> None:
        super().__post_init__()
        if self.date is None and self.total_days == self.day == self.week == 0:
            return
        if self.total_days not in range(0, 50):
            raise ValueError("Invalid Omer day (if not counting, set to 0)")
        first_omer_day = HebrewDate(month=Months.NISAN, day=16)
        last_omer_day = HebrewDate(month=Months.SIVAN, day=5)
        if self.date:
            if not first_omer_day <= self.date <= last_omer_day:
                self.total_days = 0
                self.day = 0
                self.week = 0
            else:
                first_omer_day.year = self.date.year
                self.total_days = (self.date - first_omer_day).days + 1
                self.week, self.day = divmod(self.total_days, 7)
        elif self.total_days > 0:
            self.date = first_omer_day + timedelta(days=self.total_days + 1)
            self.week, self.day = divmod(self.total_days, 7)
        else:
            self.total_days = self.week * 7 + self.day
            self.date = first_omer_day + timedelta(days=self.total_days + 1)

    def __str__(self) -> str:
        if self.total_days == 0:
            return ""
        if self.nusach == Nusach.ASHKENAZ:
            suffix = self.get_translation(f"in_omer_{self.nusach}")
        else:
            suffix = self.get_translation("in_omer")
        return f"{hebrew_number(self.total_days)} {suffix}"

    def count_str(self) -> str:
        """Return the text to be said when counting the omer."""
        if self.total_days == 0:
            return ""

        def num2words_omer(number: int, _type: str = "total") -> str:
            """Wrapper for num2words."""
            if _type == "total":
                to = "ordinal"
                type_name = "day"
                if self.language == "hebrew" and number in range(2, 11):
                    type_name = "days"
            else:
                to = "cardinal"
                type_name = _type if number == 1 else f"{_type}s"

            if self.language == "hebrew":
                conv = lang_HE.Num2Word_HE()
                construct = number == 2
                if number > 20 and number % 10 != 0:
                    count_ones = conv.to_cardinal(number % 10, gender="m")
                    count_tens = conv.to_cardinal((number // 10) * 10, gender="m")
                    count = f"{count_ones} ×•{count_tens}"
                else:
                    count = conv.to_cardinal(number, gender="m", construct=construct)
                _obj = self.get_translation(type_name)
                return f"{count} {_obj}" if number > 1 else f"{_obj} {count}"
            _obj = self.get_translation(type_name)
            count = num2words(number, lang=self.language[:2], to=to)
            if self.language == "english" and _type == "total":
                count = f"the {count}"
            if self.language == "french" and number == 1 and type_name == "week":
                count = f"{count}e"
            return f"{count} {_obj}"

        total_days = num2words_omer(self.total_days, _type="total")
        in_omer = (
            self.get_translation("in_omer")
            if self.nusach != Nusach.ASHKENAZ
            else self.get_translation("in_omer_ashkenaz")
        )
        _is = self.get_translation("is")
        prefix = (
            f"{self.get_translation('today')} {_is}".strip()
            if self.nusach != Nusach.ITALIAN
            else f"{self.get_translation('today')} {in_omer} {_is}".strip()
        )
        detail = ""
        if self.week > 0:
            which_are = self.get_translation("which_are")
            weeks = num2words_omer(self.week, _type="week")
            detail = f" {which_are} {weeks}"
            detail = f",{detail}" if self.language != "hebrew" else detail
            if self.day > 0:
                _and = self.get_translation("and")
                _and = f"{_and} " if self.language != "hebrew" else _and
                days = num2words_omer(self.day, _type="day")
                detail = f"{detail} {_and}{days}"
        if self.nusach == Nusach.ITALIAN:
            return f"{prefix} {total_days}{detail}"
        if self.nusach == Nusach.ADOT_MIZRAH:
            return f"{prefix} {total_days} {in_omer}{detail}"
        return f"{prefix} {total_days}{detail} {in_omer}"
