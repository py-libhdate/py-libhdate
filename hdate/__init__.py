"""
Jewish calendrical date and times for a given location.

HDate calculates and generates a representation either in English or Hebrew
of the Jewish calendrical date and times for a given location
"""
from hdate.common import HebrewDate, Location
from hdate.date import HDate
from hdate.htables import HolidayTypes
from hdate.zmanim import Zmanim

__all__ = ["HDate", "Zmanim", "HebrewDate", "Location", "HolidayTypes"]
