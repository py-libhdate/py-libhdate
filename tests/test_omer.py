"""Tests relating to Sefirat HaOmer."""

import random

import pytest
from syrupy.assertion import SnapshotAssertion

from hdate.omer import Omer

OMER_STRINGS = [
    (1, "היום יום אחד לעומר"),
    (2, "היום שני ימים לעומר"),
    (3, "היום שלושה ימים לעומר"),
    (7, "היום שבעה ימים שהם שבוע אחד לעומר"),
    (8, "היום שמונה ימים שהם שבוע אחד ויום אחד לעומר"),
    (10, "היום עשרה ימים שהם שבוע אחד ושלושה ימים לעומר"),
    (13, "היום שלושה עשר יום שהם שבוע אחד ושישה ימים לעומר"),
    (14, "היום ארבעה עשר יום שהם שני שבועות לעומר"),
    (17, "היום שבעה עשר יום שהם שני שבועות ושלושה ימים לעומר"),
    (19, "היום תשעה עשר יום שהם שני שבועות וחמישה ימים לעומר"),
    (28, "היום שמונה ועשרים יום שהם ארבעה שבועות לעומר"),
    (30, "היום שלושים יום שהם ארבעה שבועות ושני ימים לעומר"),
    (37, "היום שבעה ושלושים יום שהם חמישה שבועות ושני ימים לעומר"),
    (45, "היום חמישה וארבעים יום שהם שישה שבועות ושלושה ימים לעומר"),
    (49, "היום תשעה וארבעים יום שהם שבעה שבועות לעומר"),
]


@pytest.mark.parametrize("omer_day,hebrew_string", OMER_STRINGS)
def test_get_omer_string(omer_day: int, hebrew_string: str) -> None:
    """Test the value returned by calculating the Omer string."""
    assert Omer(total_days=omer_day, language="hebrew").count_str() == hebrew_string


@pytest.mark.parametrize("omer_day", range(1, 50))
@pytest.mark.parametrize("language", ["hebrew", "english", "french"])
def test_get_omer(omer_day: int, language: str, snapshot: SnapshotAssertion) -> None:
    """Test the value returned by calculating the Omer."""
    omer = Omer(total_days=omer_day, language=language)
    assert omer.count_str() == snapshot


def test_illegal_value() -> None:
    """Test passing illegal values to Omer."""
    with pytest.raises(ValueError):
        Omer(total_days=random.randint(50, 100))
    with pytest.raises(ValueError):
        Omer(total_days=random.randint(-100, -1))
