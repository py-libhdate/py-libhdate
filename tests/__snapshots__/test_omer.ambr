# serializer version: 1
# name: test_get_omer[english-10]
  'Today is the tenth day, which are one week and three days of the Omer'
# ---
# name: test_get_omer[english-11]
  'Today is the eleventh day, which are one week and four days of the Omer'
# ---
# name: test_get_omer[english-12]
  'Today is the twelfth day, which are one week and five days of the Omer'
# ---
# name: test_get_omer[english-13]
  'Today is the thirteenth day, which are one week and six days of the Omer'
# ---
# name: test_get_omer[english-14]
  'Today is the fourteenth day, which are two weeks of the Omer'
# ---
# name: test_get_omer[english-15]
  'Today is the fifteenth day, which are two weeks and one day of the Omer'
# ---
# name: test_get_omer[english-16]
  'Today is the sixteenth day, which are two weeks and two days of the Omer'
# ---
# name: test_get_omer[english-17]
  'Today is the seventeenth day, which are two weeks and three days of the Omer'
# ---
# name: test_get_omer[english-18]
  'Today is the eighteenth day, which are two weeks and four days of the Omer'
# ---
# name: test_get_omer[english-19]
  'Today is the nineteenth day, which are two weeks and five days of the Omer'
# ---
# name: test_get_omer[english-1]
  'Today is the first day of the Omer'
# ---
# name: test_get_omer[english-20]
  'Today is the twentieth day, which are two weeks and six days of the Omer'
# ---
# name: test_get_omer[english-21]
  'Today is the twenty-first day, which are three weeks of the Omer'
# ---
# name: test_get_omer[english-22]
  'Today is the twenty-second day, which are three weeks and one day of the Omer'
# ---
# name: test_get_omer[english-23]
  'Today is the twenty-third day, which are three weeks and two days of the Omer'
# ---
# name: test_get_omer[english-24]
  'Today is the twenty-fourth day, which are three weeks and three days of the Omer'
# ---
# name: test_get_omer[english-25]
  'Today is the twenty-fifth day, which are three weeks and four days of the Omer'
# ---
# name: test_get_omer[english-26]
  'Today is the twenty-sixth day, which are three weeks and five days of the Omer'
# ---
# name: test_get_omer[english-27]
  'Today is the twenty-seventh day, which are three weeks and six days of the Omer'
# ---
# name: test_get_omer[english-28]
  'Today is the twenty-eighth day, which are four weeks of the Omer'
# ---
# name: test_get_omer[english-29]
  'Today is the twenty-ninth day, which are four weeks and one day of the Omer'
# ---
# name: test_get_omer[english-2]
  'Today is the second day of the Omer'
# ---
# name: test_get_omer[english-30]
  'Today is the thirtieth day, which are four weeks and two days of the Omer'
# ---
# name: test_get_omer[english-31]
  'Today is the thirty-first day, which are four weeks and three days of the Omer'
# ---
# name: test_get_omer[english-32]
  'Today is the thirty-second day, which are four weeks and four days of the Omer'
# ---
# name: test_get_omer[english-33]
  'Today is the thirty-third day, which are four weeks and five days of the Omer'
# ---
# name: test_get_omer[english-34]
  'Today is the thirty-fourth day, which are four weeks and six days of the Omer'
# ---
# name: test_get_omer[english-35]
  'Today is the thirty-fifth day, which are five weeks of the Omer'
# ---
# name: test_get_omer[english-36]
  'Today is the thirty-sixth day, which are five weeks and one day of the Omer'
# ---
# name: test_get_omer[english-37]
  'Today is the thirty-seventh day, which are five weeks and two days of the Omer'
# ---
# name: test_get_omer[english-38]
  'Today is the thirty-eighth day, which are five weeks and three days of the Omer'
# ---
# name: test_get_omer[english-39]
  'Today is the thirty-ninth day, which are five weeks and four days of the Omer'
# ---
# name: test_get_omer[english-3]
  'Today is the third day of the Omer'
# ---
# name: test_get_omer[english-40]
  'Today is the fortieth day, which are five weeks and five days of the Omer'
# ---
# name: test_get_omer[english-41]
  'Today is the forty-first day, which are five weeks and six days of the Omer'
# ---
# name: test_get_omer[english-42]
  'Today is the forty-second day, which are six weeks of the Omer'
# ---
# name: test_get_omer[english-43]
  'Today is the forty-third day, which are six weeks and one day of the Omer'
# ---
# name: test_get_omer[english-44]
  'Today is the forty-fourth day, which are six weeks and two days of the Omer'
# ---
# name: test_get_omer[english-45]
  'Today is the forty-fifth day, which are six weeks and three days of the Omer'
# ---
# name: test_get_omer[english-46]
  'Today is the forty-sixth day, which are six weeks and four days of the Omer'
# ---
# name: test_get_omer[english-47]
  'Today is the forty-seventh day, which are six weeks and five days of the Omer'
# ---
# name: test_get_omer[english-48]
  'Today is the forty-eighth day, which are six weeks and six days of the Omer'
# ---
# name: test_get_omer[english-49]
  'Today is the forty-ninth day, which are seven weeks of the Omer'
# ---
# name: test_get_omer[english-4]
  'Today is the fourth day of the Omer'
# ---
# name: test_get_omer[english-5]
  'Today is the fifth day of the Omer'
# ---
# name: test_get_omer[english-6]
  'Today is the sixth day of the Omer'
# ---
# name: test_get_omer[english-7]
  'Today is the seventh day, which are one week of the Omer'
# ---
# name: test_get_omer[english-8]
  'Today is the eighth day, which are one week and one day of the Omer'
# ---
# name: test_get_omer[english-9]
  'Today is the ninth day, which are one week and two days of the Omer'
# ---
# name: test_get_omer[french-10]
  "Aujourd'hui c'est le dixième jour, ce qui fait une semaine et trois jours de l'Omer"
# ---
# name: test_get_omer[french-11]
  "Aujourd'hui c'est le onzième jour, ce qui fait une semaine et quatre jours de l'Omer"
# ---
# name: test_get_omer[french-12]
  "Aujourd'hui c'est le douzième jour, ce qui fait une semaine et cinq jours de l'Omer"
# ---
# name: test_get_omer[french-13]
  "Aujourd'hui c'est le treizième jour, ce qui fait une semaine et six jours de l'Omer"
# ---
# name: test_get_omer[french-14]
  "Aujourd'hui c'est le quatorzième jour, ce qui fait deux semaines de l'Omer"
# ---
# name: test_get_omer[french-15]
  "Aujourd'hui c'est le quinzième jour, ce qui fait deux semaines et un jour de l'Omer"
# ---
# name: test_get_omer[french-16]
  "Aujourd'hui c'est le seizième jour, ce qui fait deux semaines et deux jours de l'Omer"
# ---
# name: test_get_omer[french-17]
  "Aujourd'hui c'est le dix-septième jour, ce qui fait deux semaines et trois jours de l'Omer"
# ---
# name: test_get_omer[french-18]
  "Aujourd'hui c'est le dix-huitième jour, ce qui fait deux semaines et quatre jours de l'Omer"
# ---
# name: test_get_omer[french-19]
  "Aujourd'hui c'est le dix-neuvième jour, ce qui fait deux semaines et cinq jours de l'Omer"
# ---
# name: test_get_omer[french-1]
  "Aujourd'hui c'est le premier jour de l'Omer"
# ---
# name: test_get_omer[french-20]
  "Aujourd'hui c'est le vingtième jour, ce qui fait deux semaines et six jours de l'Omer"
# ---
# name: test_get_omer[french-21]
  "Aujourd'hui c'est le vingt et unième jour, ce qui fait trois semaines de l'Omer"
# ---
# name: test_get_omer[french-22]
  "Aujourd'hui c'est le vingt-deuxième jour, ce qui fait trois semaines et un jour de l'Omer"
# ---
# name: test_get_omer[french-23]
  "Aujourd'hui c'est le vingt-troisième jour, ce qui fait trois semaines et deux jours de l'Omer"
# ---
# name: test_get_omer[french-24]
  "Aujourd'hui c'est le vingt-quatrième jour, ce qui fait trois semaines et trois jours de l'Omer"
# ---
# name: test_get_omer[french-25]
  "Aujourd'hui c'est le vingt-cinquième jour, ce qui fait trois semaines et quatre jours de l'Omer"
# ---
# name: test_get_omer[french-26]
  "Aujourd'hui c'est le vingt-sixième jour, ce qui fait trois semaines et cinq jours de l'Omer"
# ---
# name: test_get_omer[french-27]
  "Aujourd'hui c'est le vingt-septième jour, ce qui fait trois semaines et six jours de l'Omer"
# ---
# name: test_get_omer[french-28]
  "Aujourd'hui c'est le vingt-huitième jour, ce qui fait quatre semaines de l'Omer"
# ---
# name: test_get_omer[french-29]
  "Aujourd'hui c'est le vingt-neuvième jour, ce qui fait quatre semaines et un jour de l'Omer"
# ---
# name: test_get_omer[french-2]
  "Aujourd'hui c'est le deuxième jour de l'Omer"
# ---
# name: test_get_omer[french-30]
  "Aujourd'hui c'est le trentième jour, ce qui fait quatre semaines et deux jours de l'Omer"
# ---
# name: test_get_omer[french-31]
  "Aujourd'hui c'est le trente et unième jour, ce qui fait quatre semaines et trois jours de l'Omer"
# ---
# name: test_get_omer[french-32]
  "Aujourd'hui c'est le trente-deuxième jour, ce qui fait quatre semaines et quatre jours de l'Omer"
# ---
# name: test_get_omer[french-33]
  "Aujourd'hui c'est le trente-troisième jour, ce qui fait quatre semaines et cinq jours de l'Omer"
# ---
# name: test_get_omer[french-34]
  "Aujourd'hui c'est le trente-quatrième jour, ce qui fait quatre semaines et six jours de l'Omer"
# ---
# name: test_get_omer[french-35]
  "Aujourd'hui c'est le trente-cinquième jour, ce qui fait cinq semaines de l'Omer"
# ---
# name: test_get_omer[french-36]
  "Aujourd'hui c'est le trente-sixième jour, ce qui fait cinq semaines et un jour de l'Omer"
# ---
# name: test_get_omer[french-37]
  "Aujourd'hui c'est le trente-septième jour, ce qui fait cinq semaines et deux jours de l'Omer"
# ---
# name: test_get_omer[french-38]
  "Aujourd'hui c'est le trente-huitième jour, ce qui fait cinq semaines et trois jours de l'Omer"
# ---
# name: test_get_omer[french-39]
  "Aujourd'hui c'est le trente-neuvième jour, ce qui fait cinq semaines et quatre jours de l'Omer"
# ---
# name: test_get_omer[french-3]
  "Aujourd'hui c'est le troisième jour de l'Omer"
# ---
# name: test_get_omer[french-40]
  "Aujourd'hui c'est le quarantième jour, ce qui fait cinq semaines et cinq jours de l'Omer"
# ---
# name: test_get_omer[french-41]
  "Aujourd'hui c'est le quarante et unième jour, ce qui fait cinq semaines et six jours de l'Omer"
# ---
# name: test_get_omer[french-42]
  "Aujourd'hui c'est le quarante-deuxième jour, ce qui fait six semaines de l'Omer"
# ---
# name: test_get_omer[french-43]
  "Aujourd'hui c'est le quarante-troisième jour, ce qui fait six semaines et un jour de l'Omer"
# ---
# name: test_get_omer[french-44]
  "Aujourd'hui c'est le quarante-quatrième jour, ce qui fait six semaines et deux jours de l'Omer"
# ---
# name: test_get_omer[french-45]
  "Aujourd'hui c'est le quarante-cinquième jour, ce qui fait six semaines et trois jours de l'Omer"
# ---
# name: test_get_omer[french-46]
  "Aujourd'hui c'est le quarante-sixième jour, ce qui fait six semaines et quatre jours de l'Omer"
# ---
# name: test_get_omer[french-47]
  "Aujourd'hui c'est le quarante-septième jour, ce qui fait six semaines et cinq jours de l'Omer"
# ---
# name: test_get_omer[french-48]
  "Aujourd'hui c'est le quarante-huitième jour, ce qui fait six semaines et six jours de l'Omer"
# ---
# name: test_get_omer[french-49]
  "Aujourd'hui c'est le quarante-neuvième jour, ce qui fait sept semaines de l'Omer"
# ---
# name: test_get_omer[french-4]
  "Aujourd'hui c'est le quatrième jour de l'Omer"
# ---
# name: test_get_omer[french-5]
  "Aujourd'hui c'est le cinquième jour de l'Omer"
# ---
# name: test_get_omer[french-6]
  "Aujourd'hui c'est le sixième jour de l'Omer"
# ---
# name: test_get_omer[french-7]
  "Aujourd'hui c'est le septième jour, ce qui fait une semaine de l'Omer"
# ---
# name: test_get_omer[french-8]
  "Aujourd'hui c'est le huitième jour, ce qui fait une semaine et un jour de l'Omer"
# ---
# name: test_get_omer[french-9]
  "Aujourd'hui c'est le neuvième jour, ce qui fait une semaine et deux jours de l'Omer"
# ---
# name: test_get_omer[hebrew-10]
  'היום עשרה ימים שהם שבוע אחד ושלושה ימים לעומר'
# ---
# name: test_get_omer[hebrew-11]
  'היום אחד עשר יום שהם שבוע אחד וארבעה ימים לעומר'
# ---
# name: test_get_omer[hebrew-12]
  'היום שנים עשר יום שהם שבוע אחד וחמישה ימים לעומר'
# ---
# name: test_get_omer[hebrew-13]
  'היום שלושה עשר יום שהם שבוע אחד ושישה ימים לעומר'
# ---
# name: test_get_omer[hebrew-14]
  'היום ארבעה עשר יום שהם שני שבועות לעומר'
# ---
# name: test_get_omer[hebrew-15]
  'היום חמישה עשר יום שהם שני שבועות ויום אחד לעומר'
# ---
# name: test_get_omer[hebrew-16]
  'היום שישה עשר יום שהם שני שבועות ושני ימים לעומר'
# ---
# name: test_get_omer[hebrew-17]
  'היום שבעה עשר יום שהם שני שבועות ושלושה ימים לעומר'
# ---
# name: test_get_omer[hebrew-18]
  'היום שמונה עשר יום שהם שני שבועות וארבעה ימים לעומר'
# ---
# name: test_get_omer[hebrew-19]
  'היום תשעה עשר יום שהם שני שבועות וחמישה ימים לעומר'
# ---
# name: test_get_omer[hebrew-1]
  'היום יום אחד לעומר'
# ---
# name: test_get_omer[hebrew-20]
  'היום עשרים יום שהם שני שבועות ושישה ימים לעומר'
# ---
# name: test_get_omer[hebrew-21]
  'היום אחד ועשרים יום שהם שלושה שבועות לעומר'
# ---
# name: test_get_omer[hebrew-22]
  'היום שניים ועשרים יום שהם שלושה שבועות ויום אחד לעומר'
# ---
# name: test_get_omer[hebrew-23]
  'היום שלושה ועשרים יום שהם שלושה שבועות ושני ימים לעומר'
# ---
# name: test_get_omer[hebrew-24]
  'היום ארבעה ועשרים יום שהם שלושה שבועות ושלושה ימים לעומר'
# ---
# name: test_get_omer[hebrew-25]
  'היום חמישה ועשרים יום שהם שלושה שבועות וארבעה ימים לעומר'
# ---
# name: test_get_omer[hebrew-26]
  'היום שישה ועשרים יום שהם שלושה שבועות וחמישה ימים לעומר'
# ---
# name: test_get_omer[hebrew-27]
  'היום שבעה ועשרים יום שהם שלושה שבועות ושישה ימים לעומר'
# ---
# name: test_get_omer[hebrew-28]
  'היום שמונה ועשרים יום שהם ארבעה שבועות לעומר'
# ---
# name: test_get_omer[hebrew-29]
  'היום תשעה ועשרים יום שהם ארבעה שבועות ויום אחד לעומר'
# ---
# name: test_get_omer[hebrew-2]
  'היום שני ימים לעומר'
# ---
# name: test_get_omer[hebrew-30]
  'היום שלושים יום שהם ארבעה שבועות ושני ימים לעומר'
# ---
# name: test_get_omer[hebrew-31]
  'היום אחד ושלושים יום שהם ארבעה שבועות ושלושה ימים לעומר'
# ---
# name: test_get_omer[hebrew-32]
  'היום שניים ושלושים יום שהם ארבעה שבועות וארבעה ימים לעומר'
# ---
# name: test_get_omer[hebrew-33]
  'היום שלושה ושלושים יום שהם ארבעה שבועות וחמישה ימים לעומר'
# ---
# name: test_get_omer[hebrew-34]
  'היום ארבעה ושלושים יום שהם ארבעה שבועות ושישה ימים לעומר'
# ---
# name: test_get_omer[hebrew-35]
  'היום חמישה ושלושים יום שהם חמישה שבועות לעומר'
# ---
# name: test_get_omer[hebrew-36]
  'היום שישה ושלושים יום שהם חמישה שבועות ויום אחד לעומר'
# ---
# name: test_get_omer[hebrew-37]
  'היום שבעה ושלושים יום שהם חמישה שבועות ושני ימים לעומר'
# ---
# name: test_get_omer[hebrew-38]
  'היום שמונה ושלושים יום שהם חמישה שבועות ושלושה ימים לעומר'
# ---
# name: test_get_omer[hebrew-39]
  'היום תשעה ושלושים יום שהם חמישה שבועות וארבעה ימים לעומר'
# ---
# name: test_get_omer[hebrew-3]
  'היום שלושה ימים לעומר'
# ---
# name: test_get_omer[hebrew-40]
  'היום ארבעים יום שהם חמישה שבועות וחמישה ימים לעומר'
# ---
# name: test_get_omer[hebrew-41]
  'היום אחד וארבעים יום שהם חמישה שבועות ושישה ימים לעומר'
# ---
# name: test_get_omer[hebrew-42]
  'היום שניים וארבעים יום שהם שישה שבועות לעומר'
# ---
# name: test_get_omer[hebrew-43]
  'היום שלושה וארבעים יום שהם שישה שבועות ויום אחד לעומר'
# ---
# name: test_get_omer[hebrew-44]
  'היום ארבעה וארבעים יום שהם שישה שבועות ושני ימים לעומר'
# ---
# name: test_get_omer[hebrew-45]
  'היום חמישה וארבעים יום שהם שישה שבועות ושלושה ימים לעומר'
# ---
# name: test_get_omer[hebrew-46]
  'היום שישה וארבעים יום שהם שישה שבועות וארבעה ימים לעומר'
# ---
# name: test_get_omer[hebrew-47]
  'היום שבעה וארבעים יום שהם שישה שבועות וחמישה ימים לעומר'
# ---
# name: test_get_omer[hebrew-48]
  'היום שמונה וארבעים יום שהם שישה שבועות ושישה ימים לעומר'
# ---
# name: test_get_omer[hebrew-49]
  'היום תשעה וארבעים יום שהם שבעה שבועות לעומר'
# ---
# name: test_get_omer[hebrew-4]
  'היום ארבעה ימים לעומר'
# ---
# name: test_get_omer[hebrew-5]
  'היום חמישה ימים לעומר'
# ---
# name: test_get_omer[hebrew-6]
  'היום שישה ימים לעומר'
# ---
# name: test_get_omer[hebrew-7]
  'היום שבעה ימים שהם שבוע אחד לעומר'
# ---
# name: test_get_omer[hebrew-8]
  'היום שמונה ימים שהם שבוע אחד ויום אחד לעומר'
# ---
# name: test_get_omer[hebrew-9]
  'היום תשעה ימים שהם שבוע אחד ושני ימים לעומר'
# ---
